test/integration/puppetlabs/services/jruby_pool_manager/jruby_pool_int_test.clj

```diff
@@ -184,80 +159,8 @@
          (is (= 1 (.size flushed-pool)))
          (is (jruby-schemas/shutdown-poison-pill? (.borrowItem flushed-pool))))))))
 
-(deftest ^:integration hold-instance-while-pool-flush-in-progress-test
```

Comment:
I think `max-borrows-flush-while-pool-flush-in-progress-test` handles the situation you described now